#! /bin/sh /usr/share/dpatch/dpatch-run
## 36_memleak.dpatch by Petter Reinholdtsen <pere@hungry.com>

Fix minor memleak.  Patch sent upstream 2006-04-01.

--- insserv-1.08.0.orig/insserv.c
+++ insserv-1.08.0/insserv.c
@@ -1773,7 +1773,7 @@
 		}
 		if (!script_inf.default_start || script_inf.default_start == empty) {
 		    if (guess->lvls)
-			script_inf.default_start = xstrdup(lvl2str(guess->lvls));
+			script_inf.default_start = lvl2str(guess->lvls);
 		}
 	    } else {
 		/*
@@ -1795,7 +1795,7 @@
 
 			list_for_each(req, &(cur->sort.req)) {
 			    if (!strcmp(getreq(req)->serv, script_inf.provides)) {
-				script_inf.default_start = xstrdup(lvl2str(getserv(ptr)->lvls));
+				script_inf.default_start = lvl2str(getserv(ptr)->lvls);
 				break;
 			    }
 			}
